### PR TITLE
[core] Add a few missing ES additions

### PR DESCRIPTION
**Pull request by pascalduez**
Namely,

`Array.prototype.findLastIndex()` [ES2023]    
MDN https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLastIndex
Spec https://tc39.es/ecma262/multipage/indexed-collections.html#sec-array.prototype.findlastindex

`String.prototype.at()` [ES2022]    
MDN https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/at
Spec https://tc39.es/ecma262/multipage/text-processing.html#sec-string.prototype.at

`TypedArray.prototype.at()` [ES2022]    
MDN https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/at
Spec https://tc39.es/ecma262/multipage/indexed-collections.html#sec-%typedarray%.prototype.at 

`Object.hasOwn()` [ES2022]    
MDN https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn  
Spec https://tc39.es/ecma262/multipage/fundamental-objects.html#sec-object.hasown

### CODE REVIEW

**Review by SamChou19815 — CHANGES_REQUESTED** [review]
You need to update the stuff in newtests as well. You can run `./tool record`

**Comment by pascalduez** [comment]
> You need to update the stuff in newtests as well. You can run `./tool record`

Done.

**Review by SamChou19815 — CHANGES_REQUESTED** [review]
Otherwise looks good

File: newtests/autostop/test.js [hunk 1]
```diff
@@ -30,12 +30,16 @@ module.exports = (suite(
...
+         [
+           {
+             "method": "telemetry/connectionStatus",
+             "params": {
+               "isConnected": true
+             }
+           }
+         ],
+         [],
+       ),
```
[the remark this answers is not in the document]
Comment (pascalduez) [reply newtests/autostop/test.js]: And reverted ;)    
This was indeed looking suspicious.

**Review by SamChou19815 — APPROVED** [review]

**Pull request closed without merging**